webapp/backend/apiClient.js

```diff
@@ -1,38 +1,65 @@
-// apiClient.js
+const axios = require('axios');
 
-// Simulate a placeholder API client module
-const callRumorDetectionAPI = async (inputData) => {
-    const apiUrl = 'https://example.com/dummy-api'; // Replace with your dummy URL
-  
-    try {
-      // In a real scenario, you would make an actual HTTP request to an external API here
-      // For simulation purposes, let's just use a delay to mimic the network request
-      // Uncomment the following lines when connecting to a real API
-      // const response = await fetch(apiUrl, {
-      //   method: 'POST',
-      //   headers: {
-      //     'Content-Type': 'application/json',
-      //   },
-      //   body: JSON.stringify({ data: inputData }),
-      // });
-  
-      // const result = await response.json();
-  
-      // Simulate a response from the imaginary API
-      const isRumor = Math.random() < 0.7;
-      return {
-        apiUrl,
-        isRumor,
-        message: isRumor ? 'This might be a rumor.' : 'No indication of a rumor.',
-      };
-    } catch (error) {
-      // Handle API request errors (e.g., network issues)
-      console.error('Error calling imaginary API:', error);
-      throw new Error('Unable to connect to the imaginary API.');
-    }
+const callHuggingFaceAPI = async (inputData) => {
+  const apiUrl = 'https://api-inference.huggingface.co/models/AbdelrahmanMosly1/BERT-Merged-Dataset';
+  const authToken = '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';
```

Comment:
Writing the authentication token in plaintext in a public repo makes the backend vulnerable. The token should be revoked and this line in the code should be changed to reference github secrets or a `.env` file that's shared between the developers and not uploaded to the repo.

Reply:
It is already revoked since commitment